lib/beaker/hypervisor.rb

```diff
@@ -87,6 +88,19 @@ def cleanup
       nil
     end
 
+    #SSH connection method preference. Can be overwritten by hypervisor to change the order
+    def self.connection_preference
+      ['ip', 'vmhostname', 'hostname']
+    end
+
+    def self.set_ssh_connection_perefence(hosts_to_provision, hypervisor)
+      if hypervisor.connection_preference.sort == self.connection_preference.sort
+        hosts_to_provision.each{ |h| h[:ssh_connection_preference] = hypervisor.connection_preference}
+      else
+        hosts_to_provision.each{ |h| h[:ssh_connection_preference] = self.connection_preference}
```

Comment:
@rishijavia rather than defaulting here, since it's hypervisor implementors who should be hitting this, not even beaker users, this should fail hard if the hypervisor implementor went to the trouble of specifying their own ordering rather than using our default, but then got it wrong.